Builds failing on Windows due to pytest upstream change

See discussion in #1782.

A recent change between versions 3.2.5 and 3.3.0 in `pytest` has caused an error to popup in the `test_edax.py` test suite (see below). It appears that one of the files is not being released properly, and so when the temporary directory is cleaned up, Windows throws an error. 

This doesn't happen in the other two test suites that use `pytest.fixture` (`test_blockfile.py` and `test_hdf5.py`).

```
λ pytest -q hyperspy\tests\io\test_edax.py
..................                                                       [100%]E [105%]

=================================== ERRORS ====================================
__________ ERROR at teardown of TestSpdMap_061_xrf.test_spc_reading ___________

    @pytest.fixture(scope="module")
    def tmpdir():
        import zipfile
        zipf = os.path.join(MY_PATH, "edax_files.zip")
        with zipfile.ZipFile(zipf, 'r') as zipped:
            with tempfile.TemporaryDirectory() as tmp:
                zipped.extractall(tmp)
                # spd_fname = os.path.join(tmp, TEST_FILES[3])
                # print(os.listdir(tmpdir))
                # print(spd_fname)
                yield tmp
                # Force files release (required in Windows)
>               gc.collect()

hyperspy\tests\io\test_edax.py:38:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\appdata\local\programs\python\python35\lib\tempfile.py:808: in __exit__
    self.cleanup()
..\..\appdata\local\programs\python\python35\lib\tempfile.py:812: in cleanup
    _shutil.rmtree(self.name)
..\..\appdata\local\programs\python\python35\lib\shutil.py:488: in rmtree
    return _rmtree_unsafe(path, onerror)
..\..\appdata\local\programs\python\python35\lib\shutil.py:383: in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = 'C:\\Users\\jat\\AppData\\Local\\Temp\\1\\tmpy0fbmh02'
onerror = <function rmtree.<locals>.onerror at 0x0000025B3BC1B400>

    def _rmtree_unsafe(path, onerror):
        try:
            if os.path.islink(path):
                # symlinks to directories are forbidden, see bug #1669
                raise OSError("Cannot call rmtree on a symbolic link")
        except OSError:
            onerror(os.path.islink, path, sys.exc_info())
            # can't continue even if onerror hook returns
            return
        names = []
        try:
            names = os.listdir(path)
        except OSError:
            onerror(os.listdir, path, sys.exc_info())
        for name in names:
            fullname = os.path.join(path, name)
            try:
                mode = os.lstat(fullname).st_mode
            except OSError:
                mode = 0
            if stat.S_ISDIR(mode):
                _rmtree_unsafe(fullname, onerror)
            else:
                try:
>                   os.unlink(fullname)
E                   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\jat\\AppData\\Local\\Temp\\1\\tmpy0fbmh02\\single_spect.spc'

..\..\appdata\local\programs\python\python35\lib\shutil.py:381: PermissionError
```

--- a/hyperspy/tests/io/test_edax.py
+++ b/hyperspy/tests/io/test_edax.py
@@ -18,51 +18,48 @@ TEST_FILES = ('Live Map 2_Img.ipr',
               'spd_map.spc',
               'spd_map.spd')
 MY_PATH = os.path.dirname(__file__)
+TMP_DIR = tempfile.TemporaryDirectory()
 
 
-@pytest.fixture(scope="module")
-def tmpdir():
+def setup_module():
     import zipfile
     zipf = os.path.join(MY_PATH, "edax_files.zip")
     with zipfile.ZipFile(zipf, 'r') as zipped:
-        with tempfile.TemporaryDirectory() as tmp:
-            zipped.extractall(tmp)
-            # spd_fname = os.path.join(tmp, TEST_FILES[3])
-            # print(os.listdir(tmpdir))
-            # print(spd_fname)
-            yield tmp
-            # Force files release (required in Windows)
-            gc.collect()
+        zipped.extractall(TMP_DIR.name)
+        # print(TMP_DIR.name)
+        # spd_fname = os.path.join(tmp, TEST_FILES[3])
+        # print(os.listdir(TMP_DIR.name))
+        # print(spd_fname)
 
 
-@pytest.fixture(scope="module")
-def spd(tmpdir):
-    signal = load(os.path.join(tmpdir, 'spd_map.spd'))
-    yield signal
-    signal.data._mmap.close()
-
-
-@pytest.fixture(scope="module")
-def spc(tmpdir):
-    os.listdir(tmpdir)
-    return load(os.path.join(tmpdir, "single_spect.spc"))
+def teardown_module():
+    TMP_DIR.cleanup()
 
 
 class TestSpcSpectrum:
 
-    def test_data(self, spc):
-        assert np.uint32 == spc.data.dtype     # test datatype
-        assert (4096,) == spc.data.shape       # test data shape
+    @classmethod
+    def setup_class(cls):
+        cls.spc = load(os.path.join(TMP_DIR.name, "single_spect.spc"))
+
+    @classmethod
+    def teardown_class(cls):
+        del cls.spc
+        gc.collect()
+
+    def test_data(self):
+        assert np.uint32 == TestSpcSpectrum.spc.data.dtype     # test datatype
+        assert (4096,) == TestSpcSpectrum.spc.data.shape       # test data shape
         assert (
             [0, 0, 0, 0, 0, 0, 1, 2, 3, 3, 10, 4, 10, 10, 45, 87, 146, 236,
-             312, 342] == spc.data[:20].tolist())   # test 1st 20 datapoints
+             312, 342] == TestSpcSpectrum.spc.data[:20].tolist()) # test 1st 20 datapoints
 
-    def test_parameters(self, spc):
-        elements = spc.metadata.as_dictionary()['Sample']['elements']
-        sem_dict = spc.metadata.as_dictionary()[
+    def test_parameters(self):
+        elements = TestSpcSpectrum.spc.metadata.as_dictionary()['Sample']['elements']
+        sem_dict = TestSpcSpectrum.spc.metadata.as_dictionary()[
             'Acquisition_instrument']['SEM']
         eds_dict = sem_dict['Detector']['EDS']
-        signal_dict = spc.metadata.as_dictionary()['Signal']
+        signal_dict = TestSpcSpectrum.spc.metadata.as_dictionary()['Signal']
 
         # Testing SEM parameters
         assert_allclose(22, sem_dict['beam_energy'])
@@ -82,9 +79,9 @@ class TestSpcSpectrum:
         # Testing HyperSpy parameters
         assert True == signal_dict['binned']
         assert 'EDS_SEM' == signal_dict['signal_type']
-        assert isinstance(spc, signals.EDSSEMSpectrum)
+        assert isinstance(TestSpcSpectrum.spc, signals.EDSSEMSpectrum)
 
-    def test_axes(self, spc):
+    def test_axes(self):
         spc_ax_manager = {'axis-0': {'name': 'Energy',
                                      'navigate': False,
                                      'offset': 0.0,
@@ -92,14 +89,23 @@ class TestSpcSpectrum:
                                      'size': 4096,
                                      'units': 'keV'}}
         assert (spc_ax_manager ==
-                spc.axes_manager.as_dictionary())
+                TestSpcSpectrum.spc.axes_manager.as_dictionary())
 
 
 class TestSpdMap:
 
-    def test_data(self, spd):
-        assert np.uint16 == spd.data.dtype     # test d_type
-        assert (200, 256, 2500) == spd.data.shape  # test d_shape
+    @classmethod
+    def setup_class(cls):
+        cls.spd = load(os.path.join(TMP_DIR.name, "spd_map.spd"))
+
+    @classmethod
+    def teardown_class(cls):
+        del cls.spd
+        gc.collect()
+
+    def test_data(self):
+        assert np.uint16 == TestSpdMap.spd.data.dtype     # test d_type
+        assert (200, 256, 2500) == TestSpdMap.spd.data.shape  # test d_shape
         assert ([[[0, 0, 0, 0, 0],              # test random data
                   [0, 0, 1, 0, 1],
                   [0, 0, 0, 0, 0],
@@ -125,14 +131,14 @@ class TestSpdMap:
                   [0, 0, 1, 0, 1],
                   [0, 0, 0, 1, 0],
                   [0, 0, 0, 0, 0]]] ==
-                spd.data[15:20, 15:20, 15:20].tolist())
+                TestSpdMap.spd.data[15:20, 15:20, 15:20].tolist())
 
-    def test_parameters(self, spd):
-        elements = spd.metadata.as_dictionary()['Sample']['elements']
-        sem_dict = spd.metadata.as_dictionary()[
+    def test_parameters(self):
+        elements = TestSpdMap.spd.metadata.as_dictionary()['Sample']['elements']
+        sem_dict = TestSpdMap.spd.metadata.as_dictionary()[
             'Acquisition_instrument']['SEM']
         eds_dict = sem_dict['Detector']['EDS']
-        signal_dict = spd.metadata.as_dictionary()['Signal']
+        signal_dict = TestSpdMap.spd.metadata.as_dictionary()['Signal']
 
         # Testing SEM parameters
         assert_allclose(22, sem_dict['beam_energy'])
@@ -152,9 +158,9 @@ class TestSpdMap:
         # Testing HyperSpy parameters
         assert True == signal_dict['binned']
         assert 'EDS_SEM' == signal_dict['signal_type']
-        assert isinstance(spd, signals.EDSSEMSpectrum)
+        assert isinstance(TestSpdMap.spd, signals.EDSSEMSpectrum)
 
-    def test_axes(self, spd):
+    def test_axes(self):
         spd_ax_manager = {'axis-0': {'name': 'y',
                                      'navigate': True,
                                      'offset': 0.0,
@@ -174,19 +180,19 @@ class TestSpdMap:
                                      'size': 2500,
                                      'units': 'keV'}}
         assert (spd_ax_manager ==
-                spd.axes_manager.as_dictionary())
+                TestSpdMap.spd.axes_manager.as_dictionary())
 
-    def test_ipr_reading(self, spd):
-        ipr_header = spd.original_metadata['ipr_header']
+    def test_ipr_reading(self):
+        ipr_header = TestSpdMap.spd.original_metadata['ipr_header']
         assert_allclose(0.014235896, ipr_header['mppX'])
         assert_allclose(0.014227346, ipr_header['mppY'])
 
-    def test_spc_reading(self, spd):
+    def test_spc_reading(self):
         # Test to make sure that spc metadata matches spd metadata
-        spc_header = spd.original_metadata['spc_header']
+        spc_header = TestSpdMap.spd.original_metadata['spc_header']
 
-        elements = spd.metadata.as_dictionary()['Sample']['elements']
-        sem_dict = spd.metadata.as_dictionary()[
+        elements = TestSpdMap.spd.metadata.as_dictionary()['Sample']['elements']
+        sem_dict = TestSpdMap.spd.metadata.as_dictionary()[
             'Acquisition_instrument']['SEM']
         eds_dict = sem_dict['Detector']['EDS']
 
@@ -199,7 +205,7 @@ class TestSpdMap:
         assert_allclose(spc_header.liveTime,
                         eds_dict['live_time'])
         assert_allclose(spc_header.evPerChan,
-                        spd.axes_manager[2].scale * 1000)
+                        TestSpdMap.spd.axes_manager[2].scale * 1000)
         assert_allclose(spc_header.kV,
                         sem_dict['beam_energy'])
         assert_allclose(spc_header.numElem,

--- a/hyperspy/tests/io/test_edax.py
+++ b/hyperspy/tests/io/test_edax.py
@@ -5,18 +5,10 @@ import gc
 
 import numpy as np
 from numpy.testing import assert_allclose, assert_equal
-import pytest
 
 from hyperspy.io import load
 from hyperspy import signals
 
-TEST_FILES = ('Live Map 2_Img.ipr',
-              'single_spect.spc',
-              'spd_map.spc',
-              'spd_map.spd',
-              'Garnet1_Img.ipr',
-              'spc0_61-ipr333.spc',
-              'spc0_61-ipr333.spd')
 MY_PATH = os.path.dirname(__file__)
 TMP_DIR = tempfile.TemporaryDirectory()
 
@@ -27,7 +19,6 @@ def setup_module():
     with zipfile.ZipFile(zipf, 'r') as zipped:
         zipped.extractall(TMP_DIR.name)
         # print(TMP_DIR.name)
-        # spd_fname = os.path.join(tmp, TEST_FILES[3])
         # print(os.listdir(TMP_DIR.name))
         # print(spd_fname)
 
@@ -36,20 +27,13 @@ def teardown_module():
     TMP_DIR.cleanup()
 
 
-@pytest.fixture(scope="module")
-def spd_061_xrf(tmpdir):
-    signal = load(os.path.join(tmpdir, 'spc0_61-ipr333_xrf.spd'))
-    yield signal
-    signal.data._mmap.close()
-
-
 class TestSpcSpectrum_v061_xrf:
 
     @classmethod
     def setup_class(cls):
-        cls.spc = load(os.path.join(TMP_DIR.name, "spc0_61-ipr333.spc"))
+        cls.spc = load(os.path.join(TMP_DIR.name, "spc0_61-ipr333_xrf.spc"))
         cls.spc_loadAll = load(os.path.join(TMP_DIR.name,
-                                            "spc0_61-ipr333.spc"),
+                                            "spc0_61-ipr333_xrf.spc"),
                                load_all_spc=True)
 
     @classmethod
@@ -329,16 +313,18 @@ class TestSpdMap_061_xrf:
 
     @classmethod
     def setup_class(cls):
-        cls.spd = load(os.path.join(TMP_DIR.name, "spc0_61-ipr333.spd"))
+        cls.spd = load(os.path.join(TMP_DIR.name, "spc0_61-ipr333_xrf.spd"))
 
     @classmethod
     def teardown_class(cls):
         del cls.spd
         gc.collect()
-        
+
     def test_data(self):
-        assert np.uint16 == TestSpdMap_061_xrf.spd.data.dtype     # test d_type
-        assert (200, 256, 2000) == TestSpdMap_061_xrf.spd.data.shape  # test d_shape
+        # test d_type
+        assert np.uint16 == TestSpdMap_061_xrf.spd.data.dtype
+        # test d_shape
+        assert (200, 256, 2000) == TestSpdMap_061_xrf.spd.data.shape
         assert ([[[0, 0, 0, 0, 0],
                   [0, 0, 0, 0, 0],
                   [0, 0, 0, 0, 0],
@@ -367,7 +353,8 @@ class TestSpdMap_061_xrf:
                 TestSpdMap_061_xrf.spd.data[15:20, 15:20, 15:20].tolist())
 
     def test_parameters(self):
-        elements = TestSpdMap_061_xrf.spd.metadata.as_dictionary()['Sample']['elements']
+        elements = TestSpdMap_061_xrf.spd.metadata.as_dictionary()['Sample'][
+            'elements']
         sem_dict = TestSpdMap_061_xrf.spd.metadata.as_dictionary()[
             'Acquisition_instrument']['SEM']
         eds_dict = sem_dict['Detector']['EDS']
@@ -424,7 +411,8 @@ class TestSpdMap_061_xrf:
         # Test to make sure that spc metadata matches spd_061_xrf metadata
         spc_header = TestSpdMap_061_xrf.spd.original_metadata['spc_header']
 
-        elements = TestSpdMap_061_xrf.spd.metadata.as_dictionary()['Sample']['elements']
+        elements = TestSpdMap_061_xrf.spd.metadata.as_dictionary()['Sample'][
+            'elements']
         sem_dict = TestSpdMap_061_xrf.spd.metadata.as_dictionary()[
             'Acquisition_instrument']['SEM']
         eds_dict = sem_dict['Detector']['EDS']

